modules/apps/frontend-js/frontend-js-web/src/main/java/com/liferay/frontend/js/web/internal/resource/handler/JavaScriptFrontendResourceRequestHandler.java

```diff
@@ -0,0 +1,218 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.js.web.internal.resource.handler;
+
+import com.liferay.frontend.js.web.internal.configuration.FrontendCachingConfiguration;
+import com.liferay.frontend.js.web.internal.resource.FrontendResource;
+import com.liferay.frontend.js.web.internal.resource.JavaScriptFrontendResource;
+import com.liferay.petra.string.CharPool;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.frontend.hashed.files.HashedFilesRegistry;
+import com.liferay.portal.kernel.frontend.hashed.files.HashedFilesUtil;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.portlet.PortletConfigFactory;
+import com.liferay.portal.kernel.util.AggregateResourceBundle;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.language.LanguageResources;
+
+import jakarta.portlet.PortletConfig;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+
+import java.io.IOException;
+
+import java.net.URL;
+
+import java.util.Enumeration;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+public class JavaScriptFrontendResourceRequestHandler
+	implements FrontendResourceRequestHandler {
+
+	public JavaScriptFrontendResourceRequestHandler(
+		FrontendCachingConfiguration frontendCachingConfiguration,
+		HashedFilesRegistry hashedFilesRegistry, Language language,
+		PortletConfigFactory portletConfigFactory) {
+
+		_frontendCachingConfiguration = frontendCachingConfiguration;
+		_hashedFilesRegistry = hashedFilesRegistry;
+		_language = language;
+		_portletConfigFactory = portletConfigFactory;
+	}
+
+	@Override
+	public boolean canHandleRequest(HttpServletRequest httpServletRequest) {
+		String requestURI = httpServletRequest.getRequestURI();
+
+		if (!requestURI.endsWith(".js")) {
+			return false;
+		}
+
+		if (HashedFilesUtil.containsHash(requestURI)) {
+			return true;
+		}
+
+		String hashedFileURI = _hashedFilesRegistry.getHashedFileURI(
+			requestURI);
+
+		if (hashedFileURI != null) {
+			return true;
+		}
+
+		URL resourceURL = _hashedFilesRegistry.getResource(requestURI);
+
+		if (resourceURL != null) {
+			return true;
+		}
+
+		return false;
+	}
+
+	@Override
+	public FrontendResource handleRequest(HttpServletRequest httpServletRequest)
+		throws IOException, ServletException {
+
+		String requestURI = httpServletRequest.getRequestURI();
+
+		String requestHash = HashedFilesUtil.getHash(requestURI);
+
+		ResourceBundle resourceBundle = _getResourceBundle(httpServletRequest);
+
+		if (requestHash != null) {
+			if (_log.isDebugEnabled()) {
```

Comment:
I get we want to log different situations, and that's why you are adding log statements within the various `if`. Nevertheless, only the last log (line 118) looks different from the rest. Are we 100% sure the other 3 capture enough information? Perhaps the fact that resourceBundle is `null` shall be logged too?